### PR TITLE
chore: adds branching policy documentation

### PR DESCRIPTION
**Pull request by jrabbott**
### Context
Adds first draft of branching policy for the project

### CODE REVIEW

**Review by colinbull — DISMISSED** [review]

File: documentation/developers/branching-policy.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Fellow developers must then perform a code reviews before merging the checked-out branch with the main branch. The crucial thing about checked-out branches is that they are short-lived, spanning two to three days at most.
+
+The main branch should always be production-ready. That means the team should thoroughly test each code change before pushing them to the main branch. Short development cycles and automated testing enable the team to identify defects and recover from failed builds quickly, reducing the risk.
+
+One of the main benefits of this approach is that it integrates well with existing continuous integration and continuous delivery (CI/CD) services. When the team pushes each commit to the main branch, it runs automated tests (part of CI) to verify that the new changes do not break the main branch. Then, after all the tests successfully complete, the pipelines are configured to create deployments.
+
+
+## Branch naming
+
+Branches should be named using the following pattern `category/<reference>/description-in-kebab-case`. With categories;
```
[the remark this answers is not in the document]
Comment (jrabbott) [reply documentation/developers/branching-policy.md:20]: Not sure I following - that section is regarding the naming of branches. It describes the pattern to follow, along with the categories to use.

File: documentation/developers/branching-policy.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Branching Policy
+
+This document is designed to outline the current approach to branching.
+
+## Trunk based development
+
+A trunk-based development workflow has been adopted within this project - trunk-based development workflow is one of the most popular development frameworks among developer teams.
+
+This branching model allows developers to collaborate on code in a single branch called ‘trunk’, resisting any pressure to create other long-lived development branches. They therefore avoid merge hell, do not break the build, and *live happily ever after*.
```
[the remark this answers is not in the document]
Comment (jrabbott) [reply documentation/developers/branching-policy.md:9]: Yes - this paragraph is describing the approach in the abstract, hence using the "trunk" in trunk-based.

File: documentation/developers/branching-policy.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (michael-fielding) [reply documentation/developers/branching-policy.md:20]: Is reference the DevOps ticket number? Or something else

[the remark this answers is not in the document]
Comment (jrabbott) [reply documentation/developers/branching-policy.md:20]: Ah ok - I'll add a paragraph

**Review by michael-fielding — APPROVED** [review]

**Pull request merged**